### PR TITLE
Correct cross-compile and image support

### PR DESCRIPTION
**Pull request by deitch**
## Description

This PR does several changes:

* Create a `Makefile.calico` to control various building and deploying options. We cannot use `Makefile` as that is generated by the build process and therefore in `.gitignore`
* Use `Makefile.calico` to have the standard targets of `build`, `image`, `image-all`, etc., as well as `ci` and `cd`
* Modify the `build.sh` script so that it uses the standard `ARCH` from other projects. All of others, `ARCH` = "target arch to which I want a binary built". Here it was `ARCH` = "arch on which I am operating and for which I will build", unless `ARCH=all`, in which case it meant "I am on `amd64` and am building for all other archs", which was very confusing. Now `ARCH`= "target arch", and it learns what it is building on via `uname -m`.
* Create docker images that contain just the binaries for that arch.
* Move the builder dockerfiles into `builder-images/` and the deployment image dockerfiles into `docker-image/` (which aligns with what some of the other projects do).
* Ensure that `create_binaries.sh` places the binaries in the correctly named output dir, so consistently `dist/amd64/`, `dist/arm64/`, `dist/ppc64le/`, `dist/s390x/`
* Rename `Dockerfile-all` to `Dockerfile-cross`, since it is used to cross-build any images, not just all
* Create `ci` target in new `Makefile.calico` to build the image. Semaphore currently has a single job `ARCH=all ./build.sh`, which can be replaced with `make -f Makefile.calico ci` to be in line with all of our other projects, and which builds binaries _and_ docker images
* Create `cd` target in new `Makefile.calico` to push the images out to docker registry

Part of the motivation here is to make docker images for all archs, so that it can be used as a distribution mechanism for `bird`, especially as required by `calico/node`.

cc @fasaxc @tomdee

### CODE REVIEW

File: Makefile.calico [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	for r in $(DOCKER_REPOS); do docker push $$r/$(CONTAINER_BASENAME):$(IMAGETAG); done
+endif
+
+## push all supported arches
+push-all: imagetag $(addprefix sub-push-,$(VALIDARCHES))
+sub-push-%:
+	$(MYMAKE) push ARCH=$* IMAGETAG=$(IMAGETAG)
+
+## tag images for one arch
+## tag images of one arch
```
Comment (fasaxc) [inline Makefile.calico]: Comment is duplicated

**Review by fasaxc — CHANGES_REQUESTED** [review]
Just one little nit

File: Makefile.calico (hunk 1 above)
Comment (deitch) [reply Makefile.calico]: Comment is duplicated

Comment (deitch) [reply Makefile.calico]: Hahaha!

**Review by fasaxc — APPROVED** [review]

**Pull request merged**